generation/proof_utils.go

```diff
@@ -153,6 +154,33 @@ func ConvertBytesToStrings(b [][32]byte) []string {
 	return s
 }
 
+func CreateVersionedState(state interface{}) spec.VersionedBeaconState {
+	var versionedState spec.VersionedBeaconState
+
+	switch s := state.(type) {
+	case *deneb.BeaconState:
+		versionedState.Deneb = s
+		versionedState.Version = spec.DataVersionDeneb
+	case *capella.BeaconState:
+		versionedState.Capella = s
+		versionedState.Version = spec.DataVersionCapella
+	}
+	return versionedState
+}
+
+func CreateVersionedBlock(block interface{}) spec.VersionedBeaconBlock {
+	var versionedBlock spec.VersionedBeaconBlock
+	switch s := block.(type) {
+	case *deneb.BeaconBlock:
+		versionedBlock.Deneb = s
+		versionedBlock.Version = spec.DataVersionDeneb
+	case *capella.BeaconBlock:
+		versionedBlock.Capella = s
+		versionedBlock.Version = spec.DataVersionCapella
+	}
+	return versionedBlock
+}
```

Comment:
this is what I could come up with for making these functions and the actual CLI files type agnostic. Obvious downside is that this could error out if the incorrect types are passed in.